src/apps/components/AppAlerts/useAppsFailedDeliveries.ts

```diff
@@ -1,23 +1,39 @@
 import { useAppFailedPendingWebhooksLazyQuery } from "@dashboard/graphql";
 import { useHasManagedAppsPermission } from "@dashboard/hooks/useHasManagedAppsPermission";
+import { Moment } from "moment";
 import { useMemo } from "react";
 
-import { webhookFailedAttemptsCheck } from "./utils";
+import { getLatestFailedAttemptFromWebhooks } from "./utils";
 
 interface AppsFailedDeliveries {
   hasFailed: boolean;
   fetchAppsWebhooks: () => void;
+  lastFailedWebhookDate: Moment | null;
 }
 
 export const useAppsFailedDeliveries = (): AppsFailedDeliveries => {
   const { hasManagedAppsPermission } = useHasManagedAppsPermission();
 
   const [fetchAppsWebhooks, { data }] = useAppFailedPendingWebhooksLazyQuery();
 
-  const hasFailed = useMemo(
+  const lastFailedWebhookDate = useMemo(
     () =>
-      data?.apps?.edges.some(app => app.node.webhooks?.some(webhookFailedAttemptsCheck), false) ??
-      false,
+      data?.apps?.edges.reduce(
+        (acc, app) => {
+          const latestFailedAttempt = getLatestFailedAttemptFromWebhooks(app.node.webhooks ?? []);
+
+          if (!latestFailedAttempt) {
+            return acc;
+          }
+
+          if (!acc) {
+            return latestFailedAttempt.createdAt;
+          }
+
+          return latestFailedAttempt.createdAt > acc ? latestFailedAttempt : acc;
+        },
+        null as Moment | null,
+      ) ?? null,
     [data],
```

Comment:
suggestion: Let's use nested data for dependencies array, so that we don't re-calculate on unrelated changes

```suggestion
    [data?.apps?.edges],
```